### PR TITLE
[KEW-1452] Removed signing_value.

### PR DESCRIPTION
**Pull request by ernestopskins**

### CODE REVIEW

File: src/wax_contract.cpp [hunk 1]
```diff
@@ -108,18 +108,28 @@ CONTRACT WAX_CONTRACT_NAME: public contract {
                 &sig_val, sizeof(sig_val), random_value, sig_it->exponent, sig_it->modulus),
              "Could not verify signature.");
         
-        jobs_table.modify(job_it, get_self(), [&](auto& rec) {
-            rec.random_value = random_value;
-        });
-
-	checksum256 rv_hash = eosio::sha256(random_value.data(),
-          random_value.size());
+        checksum256 rv_hash = eosio::sha256(random_value.data(), random_value.size());
```
Comment (jalbiero) [inline src/wax_contract.cpp]: **eosio::** is not necessary (there is a _using namespace eosio_ at the top of the file)

File: src/wax_contract.cpp [hunk 2]
```diff
@@ -108,18 +108,28 @@ CONTRACT WAX_CONTRACT_NAME: public contract {
...
+
+        jobs_table.erase(job_it);
+    }
+
+    /**
+     * @dev Removes jobs from the jobs table. The Oracle calls on it passing a list of dangling jobs.
+     * @param job_ids A vector of jobs IDs to be removed.
+     */
+    ACTION killjobs(const std::vector<int>& job_ids) {
+        for (const auto& e : job_ids) {
```
Comment (jalbiero) [inline src/wax_contract.cpp]: Silly comment: I'd use **id** instead of **e**

File: src/wax_contract.cpp [hunk 3]
```diff
@@ -108,18 +108,28 @@ CONTRACT WAX_CONTRACT_NAME: public contract {
...
+            auto job_it = jobs_table.find(e);
+            check(job_it != jobs_table.end(), "Could not find job id.");
```
Comment (jalbiero) [inline src/wax_contract.cpp]: I asked about this in the meeting: Should we assert if the job could not be found? IIRC we agreed to ignore the error.

File: tests/contracts/randreceiver.cpp [hunk 4]
```diff
@@ -41,38 +41,35 @@ CONTRACT randreceiver: public contract {
 
         print_f("receiverand called: assoc_id=%, random_value=%\n", assoc_id, rv_string );
 
-        set_last_result(assoc_id, random_value);
```
Comment (jalbiero) [inline tests/contracts/randreceiver.cpp]: You **mustn't remove random_value** in any place of this smart contract. This test smart contract implements a callback to receive a random value. The value is stored in a table in order to be validated from a unit test. Please **undo** all changes in this file.

File: tests/src/setrand_tests.cpp [hunk 5]
```diff
@@ -52,14 +52,9 @@ BOOST_FIXTURE_TEST_CASE(happy_path, helper_tester) {
...
         action_setrand(job_id, random_value);
 
-        // If the signature validation fails, the job won't be in the table.
-        auto job = get_jobs_entry(job_id);
-        BOOST_REQUIRE_EQUAL(job.random_value, random_value);
-
         // Check if the callback "receiverand" was called
         auto result = get_results_entry();
         BOOST_REQUIRE_EQUAL(result.assoc_id, assoc_id);
-        BOOST_REQUIRE_EQUAL(result.random_value, fc::sha256::hash(random_value));
```
Comment (jalbiero) [inline tests/src/setrand_tests.cpp]: Please restore this line (see comments in _randreceiver.cpp_)

File: tests/src/wax_fixture.hpp [hunk 6]
```diff
@@ -56,9 +54,8 @@ FC_REFLECT(sigpubkey_entry , (id)(exponent)(modulus));
 struct __attribute((packed)) results_entry {
     uint64_t    id;
     uint64_t    assoc_id;
-    checksum256_type random_value;
 };
-FC_REFLECT(results_entry, (id)(assoc_id)(random_value));
```
Comment (jalbiero) [inline tests/src/wax_fixture.hpp]: Please restore this line (see comments in _randreceiver.cpp_)

File: tests/src/wax_fixture.hpp [hunk 7]
```diff
@@ -39,11 +39,9 @@ FC_REFLECT(config_entry, (name)(value));
 struct __attribute((packed)) jobs_entry {
     uint64_t    id;
     uint64_t    assoc_id;
-    uint64_t    signing_value;
```
Comment (jalbiero) [inline tests/src/wax_fixture.hpp]: You don't have to remove this line, *signing_value* is used in the jobs table.

File: tests/src/requestrand_tests.cpp [hunk 8]
```diff
@@ -37,7 +37,6 @@ BOOST_FIXTURE_TEST_CASE(happy_path, wax_fixture) {
         auto job = get_jobs_entry(0);
         BOOST_REQUIRE_EQUAL(job.id, 0);
         BOOST_REQUIRE_EQUAL(job.assoc_id, 1);
-        BOOST_REQUIRE_EQUAL(job.signing_value, signing_value);
```
Comment (jalbiero) [inline tests/src/requestrand_tests.cpp]: Please restore this line (see comments in _wax_fixture.hpp_)

File: src/wax_contract.cpp (hunk 3 above)
Comment (porkchop) [reply src/wax_contract.cpp]: yeah I would agree. If the job exists then delete. if not, ignore

File: src/wax_contract.cpp (hunk 1 above)
Comment (ernestopskins) [reply src/wax_contract.cpp]: Ok, it wasn't added as part of this PR, but yes, let's remove it.

Comment (jalbiero) [reply src/wax_contract.cpp]: Yes, I know, it came from other PR.

File: src/wax_contract.cpp (hunk 2 above)
Comment (jalbiero) [reply src/wax_contract.cpp]: What's the meaning of *e*?

File: tests/contracts/randreceiver.cpp (hunk 4 above)
Comment (jalbiero) [reply tests/contracts/randreceiver.cpp]: Please *undo* all the changes in this file.

**Pull request merged**